Code/GraphMol/QueryOps.cpp

```diff
@@ -928,5 +929,136 @@ Atom *replaceAtomWithQueryAtom(RWMol *mol, Atom *atom) {
   mol->replaceAtom(idx, &qa);
   return mol->getAtomWithIdx(idx);
 }
+
+void finalizeQueryFromDescription(
+    Queries::Query<int, Atom const *, true> *query, Atom const *owner) {
+  std::string descr = query->getDescription();
+  RDUNUSED_PARAM(owner);
+
+  if (boost::starts_with(descr, "range_")) {
+    descr = descr.substr(6);
+  } else if (boost::starts_with(descr, "less_")) {
+    descr = descr.substr(5);
+  } else if (boost::starts_with(descr, "greater_")) {
+    descr = descr.substr(8);
+  }
+
+  Queries::Query<int, Atom const *, true> *tmpQuery;
+  if (descr == "AtomRingBondCount") {
```

Comment:
I think Brian was actually complaining about a different set of literal constants. 
I agree with you here, but I'd go further: I think a general refactoring of the way the queries are handled should be done. When adding a new query type it's entirely too easy to forget to add support in places like the SMARTS writer or the Pickle reader/writer.  I created a new issue for this: #4098 so that it doesn't get lost.

Reply:
I think this is a great idea.  Does this impact this PR?

Reply:
Nope, that'll be a big one on its own

Reply:
That sounds great to me too :)